x-pack/legacy/plugins/canvas/canvas_plugin_src/renderers/time_filter/components/time_filter/time_filter.tsx

```diff
@@ -3,56 +3,84 @@
+  { start: 'now-7d', end: 'now', label: strings.getLast7DaysLabel() },
+  { start: 'now-14d', end: 'now', label: strings.getLast2WeeksLabel() },
+  { start: 'now-30d', end: 'now', label: strings.getLast30DaysLabel() },
+  { start: 'now-90d', end: 'now', label: strings.getLast90DaysLabel() },
+  { start: 'now-1y', end: 'now', label: strings.getLast1YearLabel() },
+];
+
+const customQuickRanges = AdvancedSettings.get('timepicker:quickRanges').map(
+  ({ from, to, display }: { from: string; to: string; display: string }) => ({
+    start: from,
+    end: to,
+    label: display,
+  })
+);
+
+const customDateFormat = AdvancedSettings.get('dateFormat');
 
 export interface FilterMeta {
   /** Name of datetime column to be filtered  */
   column: string;
   /** Start date string of filtered date range */
-  from: string;
+  start: string;
   /** End date string of filtered date range */
-  to: string;
+  end: string;
 }
 
 function getFilterMeta(filter: string): FilterMeta {
   const ast = fromExpression(filter);
   const column = get<string>(ast, 'chain[0].arguments.column[0]');
-  const from = get<string>(ast, 'chain[0].arguments.from[0]');
-  const to = get<string>(ast, 'chain[0].arguments.to[0]');
-  return { column, from, to };
+  const start = get<string>(ast, 'chain[0].arguments.from[0]');
+  const end = get<string>(ast, 'chain[0].arguments.to[0]');
+  return { column, start, end };
 }
 
 export interface Props {
   /** Initial value of the filter */
   filter: string;
   /** Function invoked when the filter changes */
   commit: (filter: string) => void;
-  /** Determines if compact or full-sized time picker is displayed */
-  compact?: boolean;
 }
 
-export const TimeFilter = ({ filter, commit, compact }: Props) => {
-  const setFilter = (column: string) => (from: string, to: string) => {
-    commit(`timefilter from="${from}" to=${to} column=${column}`);
+export const TimeFilter = ({ filter, commit }: Props) => {
+  const setFilter = (column: string) => ({ start, end }: OnTimeChangeProps) => {
+    commit(`timefilter from="${start}" to=${end} column=${column}`);
   };
 
-  const { column, from, to } = getFilterMeta(filter);
+  const { column, start, end } = getFilterMeta(filter);
 
-  if (compact) {
-    return <TimePickerPopover from={from} to={to} onSelect={setFilter(column)} />;
-  } else {
-    return <TimePicker from={from} to={to} onSelect={setFilter(column)} />;
-  }
+  return (
+    <div className="canvasTimeFilter">
+      <EuiSuperDatePicker
+        start={start}
+        end={end}
+        isPaused={false}
+        onTimeChange={setFilter(column)}
+        showUpdateButton={false}
+        dateFormat={customDateFormat}
+        commonlyUsedRanges={customQuickRanges.length ? customQuickRanges : defaultQuickRanges}
+      />
+    </div>
+  );
 };
 
 TimeFilter.propTypes = {
   filter: PropTypes.string.isRequired,
   commit: PropTypes.func.isRequired, // Canvas filter
-  compact: PropTypes.bool,
```

Comment:
EuiSuperDatePicker is only available in "compact" view, and there is no "expanded" view of the date picker, so it makes sense to remove `compact` as an argument from the `timeFilterControl` function. By deprecating the `compact` argument, should this go into 8.0.0 due to this breaking change?

Reply:
I'm leaving the `compact` arg on the function def for `timefilterControl` so existing workpads using the `compact` arg won't start breaking.

Reply:
Maybe double check with @KOTungseth but I think we can change the release note to not be breaking since this will degrade gracefully